design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,653 @@
+  [GKE cluster external resource]'s `.ipAllocationPolicy.useIpAliases` and
+  `.ipAllocationPolicy.createSubnetwork` fields.
+* Leverage the [GKE cluster external resource]'s default behaviour of using the
+  project's `default` VPC network if none is specified.
+* Add support for `CloudSQLInstance` managed resources and their resource claims
+  to specify the external resource's `.ipConfiguration.privateNetwork`, and
+  specify the project's `default` VPC network when creating classes.
+
+To add a little more flexibility Crossplane could:
+
+* Add support for creating a new VPC network via a Crossplane VPC `Network`
+  managed resource.
+* Add support for `CloudSQLInstance` managed resources and their resource
+  classes to specify the external resource's `.ipConfiguration.privateNetwork`,
+  and specify the external name of a `Network` managed resource.
+* Add support for `GKECluster` managed resources and their resource classes to
+  specify the [GKE cluster external resource]'s `.network` field, and specify
+  the external name of a `Network` managed resource.
+* Use the `GKECluster` managed resource's existing support for setting the
+  [GKE cluster external resource]'s `.ipAllocationPolicy.useIpAliases` and
+  `.ipAllocationPolicy.createSubnetwork` fields to ensure an appropriate VPC
+  native subnet is automatically created and used.
+
+To add a generous amount of flexibility Crossplane could:
+
+* Add support for creating a new VPC network via a Crossplane VPC `Network`
+  managed resource.
+* Add support for creating a new VPC subnetwork via a VPC `Subnetwork` managed
+  resource.
+* Add support for `CloudSQLInstance` managed resources and their resource
+  classes to specify the external resource's `.ipConfiguration.privateNetwork`,
+  and specify the external name of a `Network` managed resource.
+* Add support for `GKECluster` managed resources to specify the [GKE cluster
+  external resource]'s `.network` and `.subnetwork` fields.
+* Add support for `GKECluster` managed resources and their resource classes to
+  specify the [GKE cluster external resource]'s `.network` field and
+  `.subnetwork` fields, and specify the external names of a `Network` and a
+  `Subnetwork` managed resource.
+* Use the `GKECluster` managed resource's existing support for setting the
+  [GKE cluster external resource]'s `.ipAllocationPolicy.useIpAliases` field.
+
+High-fidelity Crossplane managed resource representations of the aforementioned
+network and subnetwork external resources would look as follows:
+
+```yaml
+---
+apiVersion: vpc.gcp.crossplane.io/v1alpha1
+kind: Network
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  description: A really cool VPC network
+  # IPv4Range puts the network in subnetwork-less 'legacy mode'
+  IPv4Range: 10.0.0.0/8
+  # The documentation says this puts the VPC network in auto-create subnet mode.
+  # In my experiments it seemed to create the VPC in legacy mode instead.
```

Comment:
this `autoCreateSubnetworks` property acting the opposite of what the documentation conveys is a bit concerning, did you see any other information on that issue or other folks running into that too?

Reply:
No, but I didn't look very hard. I presume I missed something during my brief experimentation.

In my mind it's not that important to us right now. If we expose the underlying API as a high fidelity CR then we're effectively just passing this setting up faithfully to the managed resource author, and thus passing the responsibility of understanding the API up to the managed resource author. It will become more important to Crossplane once we're trying to automatically interpret and solve the connectivity needs of our customers.